packages/editor/src/pages/Auth/index.tsx

```diff
@@ -0,0 +1,215 @@
+import React from 'react';
+import QueryString from 'query-string';
+import NodeRSA from 'node-rsa';
+
+import {
+  hideSplashScreen,
+  invokeGlobalErrorHandler,
+} from 'common/lib/utilities/splash.screen';
+import { isInternetExplorer, generateCryptoSafeRandom } from 'common/lib/utilities/misc';
+import { MessageBar, MessageBarType } from 'office-ui-fabric-react/lib/MessageBar';
+import Theme from 'common/lib/components/Theme';
+import { HostType } from '@microsoft/office-js-helpers';
+import { generateGithubLoginUrl, getProfileInfo } from '../Editor/services/github';
+import IEError from './components/IEError';
+import SomethingWentWrong from './components/SomethingWentWrong';
+import UILessCodeToTokenExchanger from './components/UILessCodeToTokenExchanger';
+import OnTokenSuccess from './components/OnTokenSuccess';
+import Dialog, { DialogType } from 'office-ui-fabric-react/lib/Dialog';
+
+const SESSION_STORAGE_AUTH_COMPLETED_PARAMETER = 'auth_completed';
+const SESSION_STORAGE_AUTH_KEY_PARAMETER = 'auth_key';
+const SESSION_STORAGE_AUTH_STATE_PARAMETER = 'auth_state';
+
+interface IProps {}
+
+interface IState {
+  isIE: boolean;
+  publicKey: NodeRSA | undefined;
+  hasCodeAndState: boolean;
+  error?: string;
+
+  encodedToken?: string;
+  username?: string;
+  fullName?: string;
+  profilePicUrl?: string;
+}
+
+interface IPossibleQueryParams {
+  // The base64-encoded public key from the taskpane, if arriving on this page the first time
+  key?: string;
+
+  // Code from GitHub, if auth is successful
+  code?: string;
+  // State from GitHub, if auth is successful
+  state?: string;
+}
+
+class AuthPage extends React.Component<IProps, IState> {
+  params: IPossibleQueryParams;
+
+  constructor(props: IProps) {
+    super(props);
+
+    const isIE = isInternetExplorer();
+
+    this.params = QueryString.parse(QueryString.extract(window.location.href));
+
+    let base64Key: string | undefined;
+    if (typeof this.params.key === 'string' && this.params.key.trim().length > 0) {
```

Comment:
isn't this always a string? or undefined?

Reply:
technically yes.  But what if someone messed it up and passed a number, which doesn't have ".trim()" on it.  I'd rather fail gracefully-ier in that case.

Reply:
I meant doesn’t querystring always return it as a string or will it try to be smart and return a number